airbyte-integrations/connectors/source-netsuite-odbc/source_netsuite_odbc/streams.py

```diff
@@ -0,0 +1,424 @@
+
+      We return state after 10,000 records because that is how large of a page we can return from Netsuite in
+      a sustainable way, and we should be able to restart successfully after each page.
+      """
+      return 10000
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+      if 'last_date_updated' in current_stream_state and self.incremental_most_recent_value_seen is not None:
+        last_date_updated = max(current_stream_state['last_date_updated'], self.incremental_most_recent_value_seen)
+      else:
+        last_date_updated = self.incremental_most_recent_value_seen
+      if 'last_values_seen' in current_stream_state:
+        last_values_seen = self.update_last_values_seen_from_current_stream_state(current_stream_state)
+      else:
+        last_values_seen = self.primary_key_last_value_seen
+
+      return {
+        'last_values_seen': last_values_seen,
+        'last_date_updated': last_date_updated
+      }
+    
+    def update_last_values_seen_from_current_stream_state(self, current_stream_state: MutableMapping[str, Any]):
+      stream_last_values = current_stream_state['last_values_seen']
+      new_values = {}
+      for key in self.primary_key_column:
+        stream_value = stream_last_values[key] if key in stream_last_values else STARTING_PRIMARY_KEY_VALUE
+        non_stream_value = self.primary_key_last_value_seen[key] if key in self.primary_key_last_value_seen else STARTING_PRIMARY_KEY_VALUE
+        new_values[key] = max(stream_value, non_stream_value)
+      return new_values
+
+    # END STATE MANAGEMENT FUNCTIONS
+
+    # QUERY GENERATION FUNCTIONS
+    def has_composite_primary_key(self):
+      return len(self.primary_key_column) > 1
+    
+    def generate_query(self, sync_mode, stream_slice):
+      if sync_mode == SyncMode.full_refresh:
+        return self.generate_full_refresh_query(stream_slice)
+      elif sync_mode == SyncMode.incremental:
+        return self.generate_incremental_query(stream_slice)
+      else:
+        raise Exception(f'Unsupported Sync Mode: {sync_mode}.  Please use either "full_refresh" or "incremental"')
+
+    # How does full refresh work?
+    # We first split things into stream slices.  Then, we start with filtering our SQL query by > the start date of the stream slice
+    # and <= the end date of the stream slice.  We then sort by the primary key column (s).
+    #  As we fetch records, we start filtering by > the most recent primary key value we've seen.  Because primary keys 
+    # can be composite, we need to handle the cases where the first column might have multiple rows for a single value (since the second
+    # or third row might have different values)
+    # Note:  Outside of the stream slice dates, we don't filter by dates in full refresh.  We instead rely on Netsuite's ids being incremental.
+    def generate_full_refresh_query(self, stream_slice):
```

Comment:
can you add a fully generated query as an example in a comment? one for full refresh and one for incremental